File: backend/models/user_model.py
class UserModel:
    def __init__(self, user_id, name, email, password_hash=None, onboarding_completed=False, age=None, gender=None, profile_image=None):
        self.user_id = user_id
        self.name = name
        self.email = email
        self.password_hash = password_hash
        self.onboarding_completed = onboarding_completed
        self.age = age
        self.gender = gender
        self.profile_image = profile_image

    def to_dict(self):
        """Convert model representation into dictionary format suitable for Firestore."""
        data = {
            'user_id': self.user_id,
            'name': self.name,
            'email': self.email,
            'onboarding_completed': self.onboarding_completed,
            'age': self.age,
            'gender': self.gender,
            'profile_image': self.profile_image
        }
        if self.password_hash:
            data['password_hash'] = self.password_hash
        return data

    @staticmethod
    def from_dict(data):
        """Instantiate user model from firestore dictionary data."""
        if not data:
            return None
        return UserModel(
            user_id=data.get('user_id'),
            name=data.get('name'),
            email=data.get('email'),
            password_hash=data.get('password_hash'),
            onboarding_completed=data.get('onboarding_completed', False),
            age=data.get('age'),
            gender=data.get('gender'),
            profile_image=data.get('profile_image')
        )

File: frontend/lib/models/user_model.dart
class UserModel {
  final String id;
  final String name;
  final String email;
  final int? age;
  final String? gender;
  final bool onboardingCompleted;

  UserModel({
    required this.id,
    required this.name,
    required this.email,
    this.age,
    this.gender,
    this.onboardingCompleted = false,
  });

  String get userId => id;

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: json['user_id'] ?? json['id'] ?? '',
      name: json['name'] ?? json['username'] ?? '',
      email: json['email'] ?? '',
      age: json['age'] is int ? json['age'] : int.tryParse(json['age']?.toString() ?? ''),
      gender: json['gender'],
      onboardingCompleted: json['onboarding_completed'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'user_id': id,
      'name': name,
      'email': email,
      'age': age,
      'gender': gender,
      'onboarding_completed': onboardingCompleted,
    };
  }
}
